server/services/offer.service.js

```diff
@@ -509,6 +520,10 @@ export const getAllUserOffers = async (user, accounId, page = 1, limit = 10) =>
     offerOptions.unshift({ $match: { vendorId: ObjectId(user._id) } });
   }
 
+  if (user.role === USER_ROLE.USER && user._id.toString() !== accounId.toString()) {
```

Comment:
Can we fix the `missing t` typo in accounId anywhere it is used
```suggestion
  if (user.role === USER_ROLE.USER && user._id.toString() !== accountId.toString()) {
```